### PR TITLE
ath79: create DTSI for ar9341 TP-Link devices

### PR DESCRIPTION
**Pull request by adschm**
This patch creates a shared DTSI for the TP-Link devices based
on ar9341 as those share a lot of definitions.
It looks like the DTSI can be used by upcoming device support
patches, too.

While at it, change from gpio-keys-polled to gpio-keys.

### CODE REVIEW

File: target/linux/ath79/dts/ar9341_tplink.dtsi [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	phy-handle = <&swphy0>;
+
+	mtd-mac-address = <&uboot 0x1fc00>;
+	mtd-mac-address-increment = <(-1)>;
+};
+
+&eth1 {
+	mtd-mac-address = <&uboot 0x1fc00>;
+
+	pll-data = <0x06000000 0x00000101 0x00001616>;
```
Comment (adrianschmutzler) [inline target/linux/ath79/dts/ar9341_tplink.dtsi]: @981213 Can this be removed also here as you implied for MR3420v2?

Comment (981213) [reply target/linux/ath79/dts/ar9341_tplink.dtsi]: Yes. pll-data is only used when there is a pll-handle defined.

**Comment by adrianschmutzler** [comment]
I've included the current state of #2317 and rebased onto that, so I can merge all three definitions.

@gwlim If you like, it would be helpful if you could test this on your device (it should not change anything compared to your patch).

**Review by gwlim — APPROVED** [review]
Tested Works on MR3420v2

**Comment by 981213** [comment]
Merged into my staging tree at: https://git.openwrt.org/?p=openwrt/staging/981213.git
Thanks!

**Pull request closed without merging**